### ISSUE TITLE
New Rule: no-ineffective-emphasis

### ISSUE THREAD

**Issue by tats-u** [eslint/markdown#395]
### Rule details

This rule warns `*` and `**` incompatible with CommonMark especially in some languages (especially Japanese and Chinese and sometimes Korean)

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
**この文は太字になりません。（日本語）**左边的句子没有加粗。（中文）

<!-- **This sentence will not be bold. (Japanese)** The left sentence will not be bold. (Chinese) -->
<!-- Asterisks in the English-translated sentences will be converted to <strong> but those in the original Japanese and Chinese sentences will NOT be -->
```

### Prior Art

```markdown

```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

This spec bug in CommonMark has suffered Japanese and Chinese Markdown users for a long time.

- https://github.com/tats-u/markdown-cjk-friendly
- https://github.com/commonmark/commonmark-spec/issues/650
- https://community.openai.com/t/markdown-bold-formatting-error-in-japanese/1111571
- https://talk.commonmark.org/t/emphasis-and-east-asian-text/2491
- https://talk.commonmark.org/t/cjk-and-bold-tags-possibly-complex-issue/4034/7
- https://talk.commonmark.org/t/bold-failure-in-chinese-sentence-because-of-the-usage-of-chinese-punctuation-is-different-from-english/3478/17

![Image](https://github.com/user-attachments/assets/0096c3e7-d15c-4711-878d-e97e1586a2a3)

![Image](https://github.com/user-attachments/assets/1d33d75a-a27f-4347-a2af-3eac115c0c22)

![Image](https://github.com/user-attachments/assets/f07e032d-c174-46a5-9d7c-35bde231330a)

![Image](https://github.com/user-attachments/assets/0711e498-4fab-481f-9feb-43c54408fffb)

↓ The 2 surrounding spaces around the bold link are essentially unnecessary, but added to avoid this problem.

![Image](https://github.com/user-attachments/assets/6dadfad2-98a6-4dfb-9436-529e1ba9860b)

**Comment by tats-u** [comment]
Never there. Underlines have no effect unless _both_ outsides are space or punctuation. Less compatible with CJK languages, especially Chinese and Japanese, than asterisks.

**Comment by tats-u** [comment]
Few valid use cases:

```md
__重要__：左は太字になります。

重要：__ここは太字になります。__

次は太字になります。__ここは太字になります__。__ここも太字になります。__
```

<div lang="ja">

__重要__：左は太字になります。

重要：__ここは太字になります。__

次は太字になります。__ここは太字になります__。__ここも太字になります。__

</div>

However, only CommonMark experts can deal with `_`'s behavior in (Chinese and) Japanese text.

**Comment by nzakas** [comment]
Thanks for the clarification. My feeling at this moment is that this would be better suited to a different plugin that is focused on internationalization issues. It's very common to use `*` and `**` in English, so I'm not sure having a rule in this plugin makes sense.

Remember, this doesn't have to be the only ESLint plugin with Markdown rules, it's just the core rules we believe are important. As with JavaScript, our intent is to let the community write the rules they feel are important.

@eslint/eslint-team what do others think?

**Comment by tats-u** [comment]
> It's very common to use * and ** in English, so I'm not sure having a rule in this plugin makes sense.

Opt-in or an additional behavioral option to restrict to CJK context is fine.

**Comment by kecrily** [comment]
As a Chinese user, I have suffered from this problem. Some parsers can handle this, some can't.


> Opt-in or an additional behavioral option to restrict to CJK context (only when a CJK character is around an asterisk or inside of the emphasized range) is fine.

The cost of CJK detection is too big for me to support.

I'd prefer this as a style rule (but I'm not sure it would be accepted at this time), with the option of using `_` instead of `*` for users who have problems.

**Comment by mdjermanovic** [comment]
I agree with @nzakas that this seems better suited to a plugin focused on internationalization/language-specific issues.

As with JS rules, anyone can write and publish a plugin with rules made to work with the core (AST, SourceCode...) provided by `@eslint/markdown`.

**Comment by fasttime** [comment]
I'm not sure I understand the extent of this problem. It seems to me that `_`, `__`, `*` and `**` only operate as emphasis tags around certain characters, and this behavior is consistent with the spec. But:
* People don't know and so they end up using emphasis tags where they have no effect.
* The spec was unclear. It was amended to clarify the intended behavior but it's still not entirely clear.
* Some parsers do not adhere to the spec accurately.

Can somebody confirm that this is correct?

**Comment by lumirlumir** [comment]
@fasttime I would say yes, as far as I understand.
